### PR TITLE
codeninja teacher can create club

### PR DESCRIPTION
**Pull request by smallst**
fix ENG-979
image
image

## Summary by CodeRabbit

- **New Features**
  - Added support for creating clubs within the application, allowing users to classify classrooms as clubs.
  - Introduced new labels for localization related to club management, enhancing user interface clarity.
  - Enhanced teacher dashboard with new functionalities for club creation and interaction, including visibility control for club-specific modals.
  - Improved modal for editing classes to support club-specific fields and options.

- **Bug Fixes**
  - Improved handling of modal states related to classroom and club creation to ensure better user experience.

### CODE REVIEW

**Comment by smallst** [comment]
image
update modal title

File: app/schemas/models/classroom.schema.js [hunk 1]
```diff
@@ -116,7 +116,7 @@ _.extend(ClassroomSchema.properties, {
   }, {}),
   stats: c.object({ additionalProperties: true }),
   initializedOuterStats: { type: 'boolean', default: true, description: 'whether the classroom.stats for the classroom is initialized' },
-  type: { title: 'Class Type', type: 'string', enum: ['', 'in-school', 'after-school', 'online', 'camp', 'camp-esports', 'camp-junior', 'homeschool', 'other'] }
+  type: { title: 'Class Type', type: 'string', enum: ['', 'in-school', 'after-school', 'online', 'camp', 'camp-esports', 'camp-junior', 'homeschool', 'other', 'club', 'club-esports', 'club-roblox', 'club-hackstack', 'club-junior'] }
```
Comment (mrfinch) [inline app/schemas/models/classroom.schema.js:119]: we only had 4 clubs: AI HS, ozaria, Roblox, esports, why is there a junior in here?  Also, let's not call it club but club-ozaria. Let's not show different options but show an error if we can't create ozaria on coco.

**Review by mrfinch — COMMENTED** [review]
I like the idea of new option of club instead of replacing the class but we will need to talk with them if that works for them.

File: app/schemas/models/classroom.schema.js (hunk 1 above)
Comment (smallst) [reply app/schemas/models/classroom.schema.js:119]: image
lol maybe we remove it later?

**Review by mrfinch — APPROVED** [review]

**Pull request merged**